File: Zadania Szkopul (C++ i Java)/Sylwester Halinki/Sylwester_Halinki.java
import java.util.Scanner;

public class Sylwester_Halinki {
    public static void main(String[] args) {
        int ta,chl; // ta - ilosc tancow; chl - ilosc chlopakow
        int p=1,k=1,tmp,wynik = -1000; // poczatek i koniec gegawicy
        int Tab[] = new int[100005];
        Scanner czytaj = new Scanner(System.in);
        ta = czytaj.nextInt();
        chl = czytaj.nextInt();
        for(int i=1; i<=ta; i++){
            Tab[i] = czytaj.nextInt();
        }
        czytaj.close();
        while(k < ta+1){
            if(Tab[k] != Tab[k+1] && Tab[k+1] != 0){
                k++;
            }
            else{
                tmp = k - p + 1;
                wynik = Math.max(tmp, wynik);
                p = k + 1;
                k = p;
            }
        }
        System.out.println(wynik);
    }
}

File: Zadania Szkopul (C++ i Java)/Sylwester Halinki/Sylwester_Halinki.cpp
#include <bits/stdc++.h>

using namespace std;
int ta,chl;
int p=1,k=1,tmp,wynik = -1000; // p, k - poczatek i koniec gasienicy; tmp - zmienna tymczasowa; wynik - koncowy wynik programu
int Tab[100005];
int main()
{
	scanf("%d %d",&ta,&chl);
	for(int i=1;i<=ta;i++)
	{
		scanf("%d", &Tab[i]);
	}
		while(k<ta+1)
		{
			if(Tab[k]!=Tab[k+1] and Tab[k+1]!=0)
				k++;
			else
			{
				tmp=k-p+1;
				wynik=max(tmp,wynik);
				p=k+1;
				k=p;
			}
		}
	printf("%d",wynik);
}